cmake/external_mlsl.cmake

```diff
@@ -0,0 +1,51 @@
+# ******************************************************************************
+# Copyright 2017-2018 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ******************************************************************************
+
+# Enable ExternalProject CMake module
+include(ExternalProject)
+
+#------------------------------------------------------------------------------
+# Download MLSL
+#------------------------------------------------------------------------------
+
+SET(MLSL_GIT_URL https://github.com/intel/MLSL)
+SET(MLSL_GIT_TAG master)
+
+find_program(MAKE_EXE NAMES gmake nmake make)
+
+ExternalProject_Add(
+    MLSL
+    PREFIX MLSL
+    GIT_REPOSITORY ${MLSL_GIT_URL}
+    GIT_TAG ${MLSL_GIT_TAG}
+    UPDATE_COMMAND ""
+    CONFIGURE_COMMAND ""
+    BUILD_COMMAND ${MAKE_EXE} -j 1
+    INSTALL_COMMAND ${MAKE_EXE} install PREFIX=${EXTERNAL_PROJECTS_ROOT}/MLSL/install
+    BUILD_IN_SOURCE TRUE
+    TMP_DIR "${EXTERNAL_PROJECTS_ROOT}/MLSL/tmp"
+    STAMP_DIR "${EXTERNAL_PROJECTS_ROOT}/MLSL/stamp"
+    SOURCE_DIR "${EXTERNAL_PROJECTS_ROOT}/MLSL/src"
+    INSTALL_DIR "${EXTERNAL_PROJECTS_ROOT}/MLSL/install"
+    EXCLUDE_FROM_ALL TRUE
+    )
+
+ExternalProject_Get_Property(MLSL SOURCE_DIR)
+ExternalProject_Get_Property(MLSL INSTALL_DIR)
+add_library(libmlsl INTERFACE)
+target_include_directories(libmlsl SYSTEM INTERFACE ${SOURCE_DIR}/include)
+target_link_libraries(libmlsl INTERFACE ${INSTALL_DIR}/intel64/lib/libmlsl.so)
+add_dependencies(libmlsl MLSL)

```

Comment:
Do we need to add an install for this library so when we do `make install`? Do a make install and verify that the libmlsl is there.

Reply:
make install is called during "make all" invocation. I've added additional cmake install commands to copy .so and header to the common install destination. I've tried to install MLSL to CMAKE_INSTALL_PREFIX directly, but MLSL install target cleans target directory